markdown.go

```diff
@@ -314,24 +319,20 @@ func MarkdownBasic(input []byte) []byte {
 //
 // * Custom Heading IDs
 func MarkdownCommon(input []byte) []byte {
-	// set up the HTML renderer
-	renderer := NewHTMLRenderer(HTMLRendererParameters{
-		Flags: CommonHTMLFlags,
-	})
-	return Markdown(input, renderer, DefaultOptions)
+	return Markdown(input, DefaultOptions)
 }
 
 // Markdown is the main rendering function.
 // It parses and renders a block of markdown-encoded text.
-// The supplied Renderer is used to format the output, and extensions dictates
-// which non-standard extensions are enabled.
+// The supplied options contain a Renderer used to format the output, and
+// extensions that dictate which non-standard extensions are enabled.
```

Comment:
This doesn't seem to make for a friendly API, as I understand it, because providing a zero value for blackfriday.Options{} doesn't do what one would expect in this style of API.

Here's the behavior I expect and think we should provide:

```Go
out := blackfriday.Markdown(in, blackfriday.Options{})
```

That should mean "use default options, nothing is overridden to custom values".

Next,

```Go
out := blackfriday.Markdown(in, blackfriday.Options{
	Extensions: extensions,
	Renderer:   renderer,
})
```

That should mean "use the provided extensions and renderer."